packages/core/src/odata/common/filter/filterable.ts

```diff
@@ -67,3 +72,40 @@ export function or<EntityT extends EntityBase>(
 // Export function not<EntityT extends Entity>(expression: Filterable<EntityT>): Filterable<EntityT> {
 //   Return new FilterList([], expressions);
 // }
+
+// eslint-disable-next-line valid-jsdoc
+/**
+ * @experimental This is experimental and is subject to change. Use with caution.
+ */
+export function any<FieldT extends FieldType, LinkedEntityT extends EntityBase>(
```

Comment:
General for `any` and `all`. When I checked the [rules](http://docs.oasis-open.org/odata/odata/v4.0/errata03/os/complete/abnf/odata-abnf-construction-rules.txt) it stated for all (any analogous):
```
allExpr = 'all' OPEN BWS   lambdaVariableExpr BWS COLON BWS lambdaPredicateExpr   BWS CLOSE
```
So you have a lambdaVariableExpr which is defined by:
```
lambdaPredicateExpr = boolCommonExpr ; containing at least one lambdaVariableExpr
```
And a boolCommonExpr can if I understand it correctly contain also a and/or condition. I am not good with this abstract definition but du you know is something like:
any(and(PropA.eq('Foo),PropB.eq('Bar')) is allowed? For it looks like it and then the argument of the any / all should be extended to allow for `FilterList`.

Reply:
Tested with Marika, syntax wise yes, you can use `FilterList`.
However, the response is strange.
e.g.,
- Friends/any(f:f/UserName eq 'scottketchum') gives you something [1] (as expected)
- Friends/any(f:f/UserName eq 'scottketchum' or UserName eq 'scottketchum') gives you the same as [1] (as expected)
- Friends/any(f:f/UserName eq 'scottketchum' and UserName eq 'scottketchum') gives nothing (???)
decided to support FilterList, since we are not doing something wrong.